### PR TITLE
Resolve ${workspaceRoot} in configFile path

### PR DESCRIPTION
**Pull request by iainmcgin**
I have a globally specified tslint file, that I occasionally wish to override on a per-workspace basis but don't want to include the full, absolute path to the workspace configuration files in my `.vscode/settings.json`. This change allows `${workspaceRoot}` to be substituted for the full path on demand.

Performing that substitution within the extension would likely be cleaner, but I don't know enough about how configuration is passed to language servers from extensions to find the appropriate hook to do this. The proposed change here instead records the workspace root during language server initialization, and then applies the substitution on configuration change events. This feels like a bit of a hack to me; if anyone can point me towards a way to intercept and modify a configuration prior to sending it to the language server, I'd be happy to put together a separate PR.

### CODE REVIEW

File: tslint-server/src/server.ts [hunk 1]
```diff
@@ -71,6 +71,7 @@ namespace StatusNotification {
 	export const type = new server.NotificationType<StatusParams, void>('tslint/status');
 }
 
+let rootDir: string | null = null;
```
Comment (egamma) [inline tslint-server/src/server.ts]: Where is the variable `rootDir` set? In the changes I only see that `rootDir` is accessed

**Comment by iainmcgin** [comment]
My user settings point to a tslint.json that is intended to be used across all of Google's TypeScript code. However I am also working on an open source project that uses a separate tslint style. This project has its tslint.json in the root, as you say. When the global file is inaccessible (e.g. when I haven't mounted the virtual file system it is on) I repeatedly see the error:

```
WARN: The file /path/to/global/tslint.json refered to by 'tslint.configFile' does not exist
```

This is occurring even when a local tslint.json file exists, so it appears that the tslint.configFile value may be taking precedence over the local file.

Having experimented a little more, it looks like setting tslint.configFile in .vscode/settings.json to "tslint.json" will automatically resolve this relative path to the right location. So, this PR is not actually needed. It may be worth including anyway for consistency with the configuration of other plugins that support ${workspaceRoot}.

File: tslint-server/src/server.ts (hunk 1 above)
Comment (iainmcgin) [inline tslint-server/src/server.ts]: Looks like I had an unstaged change :( I'll update the PR, in case you still want to accept it.

**Comment by egamma** [comment]
I now understand your scenario, thanks. You are correct that the setting tslint.configFile takes precedence in the lookup for the tslint.json when it is defined. Using a relative path in the setting works as you have already verified, so this is a good solution.

Since there is currently no support for variables in settings in general (we are discussing it), I'm closing this PR.

**Pull request closed without merging**